util/secret/secret.go

```diff
@@ -40,8 +41,17 @@ func GetFromNamespacedName(ctx context.Context, c client.Reader, clusterName cli
 		Namespace: clusterName.Namespace,
 		Name:      Name(clusterName.Name, purpose),
 	}
-
 	if err := c.Get(ctx, secretKey, secret); err != nil {
+		if purpose == UserKubeconfig {
```

Comment:
I think this logic should sit in the kubeconfig package as this is just to get a secret with a specific purpose.